<<<<<<< HEAD
$NetBSD$

Split configuration between -client and -server.
Do not run libmysql_api_test, requires LD_LIBRARY_PATH.

--- libmysql/CMakeLists.txt.orig	2019-04-13 11:46:31.000000000 +0000
+++ libmysql/CMakeLists.txt
@@ -241,11 +241,17 @@ IF(WIN32)
 ENDIF()
 
 # LDAP authentication SASL client plugin
+IF(NOT WITHOUT_SERVER)
 MESSAGE(STATUS "Creating LDAP authentication SASL client library.")
 ADD_SUBDIRECTORY(authentication_ldap)
+ENDIF()
   
=======
$NetBSD: patch-libmysql_CMakeLists.txt,v 1.1 2021/05/13 15:25:20 jdolecek Exp $

Do not install libmysqlclient with the server.
Generate versioned lib symlinks on FreeBSD to supress PLIST divergence.
authentication_ldap and libmysql_api_test require libmysqld.

--- libmysql/CMakeLists.txt.orig	2021-03-22 08:44:50.000000000 +0000
+++ libmysql/CMakeLists.txt
@@ -270,11 +270,12 @@ IF(WIN32)
   LIST(APPEND LIBS_TO_MERGE auth_win_client)
 ENDIF()
 
-# LDAP authentication SASL client plugin
-ADD_SUBDIRECTORY(authentication_ldap)
-
>>>>>>> netbsd/trunk
 # Merge several convenience libraries into one big mysqlclient
+IF(WITHOUT_SERVER)
 MERGE_CONVENIENCE_LIBRARIES(mysqlclient ${LIBS_TO_MERGE} COMPONENT Development)
+ELSE()
+MERGE_CONVENIENCE_LIBRARIES(mysqlclient SKIP_INSTALL ${LIBS_TO_MERGE} COMPONENT Development)
+ENDIF()
 TARGET_LINK_LIBRARIES(mysqlclient PRIVATE ${LIBS_TO_LINK})
 
 # Visual Studio users need debug  static library for debug projects
<<<<<<< HEAD
@@ -284,9 +290,15 @@ ENDIF()
 IF(NOT DISABLE_SHARED)
   # Merge several convenience libraries into one big mysqlclient
   # and link them together into shared library.
+  IF(WITHOUT_SERVER)
   MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
     EXPORTS ${CLIENT_API_FUNCTIONS} ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
     COMPONENT SharedLibraries)
+  ELSE()
+  MERGE_LIBRARIES_SHARED(libmysql SKIP_INSTALL ${LIBS_TO_MERGE}
+    EXPORTS ${CLIENT_API_FUNCTIONS} ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
+    COMPONENT SharedLibraries)
+  ENDIF()
   TARGET_LINK_LIBRARIES(libmysql PRIVATE ${LIBS_TO_LINK})
 
   IF(WIN32)
@@ -359,6 +371,7 @@ CONFIGURE_FILE(api_test.c.in ${CMAKE_CUR
=======
@@ -310,12 +311,21 @@ ENDIF()
 
 # Merge several convenience libraries into one big mysqlclient
 # and link them together into shared library.
+IF(WITHOUT_SERVER)
 MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
   EXPORTS
   ${CLIENT_API_FUNCTIONS}
   ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
   ${CLIENT_API_NONBLOCKING_FUNCTIONS}
   COMPONENT SharedLibraries)
+ELSE()
+MERGE_LIBRARIES_SHARED(libmysql SKIP_INSTALL ${LIBS_TO_MERGE}
+  EXPORTS
+  ${CLIENT_API_FUNCTIONS}
+  ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
+  ${CLIENT_API_NONBLOCKING_FUNCTIONS}
+  COMPONENT SharedLibraries)
+ENDIF()
 TARGET_LINK_LIBRARIES(libmysql PRIVATE ${LIBS_TO_LINK})
 
 IF(LINUX_STANDALONE AND KERBEROS_CUSTOM_LIBRARY)
@@ -334,7 +344,7 @@ ENDIF()
 
 IF(UNIX)
   # libtool compatability
-  IF(FREEBSD OR APPLE)
+  IF(APPLE)
     SET(OS_SHARED_LIB_VERSION "${SHARED_LIB_MAJOR_VERSION}")
   ELSE()
     SET(OS_SHARED_LIB_VERSION
@@ -397,6 +407,7 @@ CONFIGURE_FILE(api_test.c.in ${CMAKE_CUR
>>>>>>> netbsd/trunk
 # from @CLIENT_API_FUNCTIONS@ are declared by <mysql.h>. It will fail
 # to run if not all of these symbols are exported by the library.
 #
+IF(NOT WITHOUT_SERVER)
 MYSQL_ADD_EXECUTABLE(libmysql_api_test
<<<<<<< HEAD
   ${CMAKE_CURRENT_BINARY_DIR}/api_test.c SKIP_INSTALL)
 # Clang/UBSAN needs this on some platforms.
@@ -380,9 +393,12 @@ IF(DISABLE_SHARED)
 ELSE()
   TARGET_LINK_LIBRARIES(libmysql_api_test libmysql)
 ENDIF()
+ENDIF()
 
 # Verify that libmysql_api_test runs OK
+IF(0)
 ADD_CUSTOM_COMMAND(TARGET libmysql_api_test POST_BUILD
   COMMAND $<TARGET_FILE:libmysql_api_test>
=======
   ${CMAKE_CURRENT_BINARY_DIR}/api_test.c
   LINK_LIBRARIES libmysql ${LIBRT}
@@ -425,3 +436,4 @@ ADD_CUSTOM_COMMAND(TARGET libmysql_api_t
   COMMAND libmysql_api_test
>>>>>>> netbsd/trunk
   > ${CMAKE_CURRENT_BINARY_DIR}/libmysql_api_test.out
   )
+ENDIF()
